### PR TITLE
Fix gnmi cli hang

### PR DESCRIPTION
**Pull request by zbud-msft**
#### Why I did it

gnmi_cli will hang when trying to write to channel inside cli_config. Use one goroutine to write to term channel and check if received cnt == expected cnt, otherwise when sleep_cnt hits max term due to timeout

#### How to verify it

Nightly tests

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: patches/0001-Fix-gnmi_cli-hang.patch [hunk 1]
```diff
@@ -0,0 +1,47 @@
+From be4d7a0f0c902fc5fc7cc4dbc2f8342679b4daf9 Mon Sep 17 00:00:00 2001
+From: zbud-msft <zainbudhwani@microsoft.com>
+Date: Wed, 4 Jan 2023 00:07:04 +0000
+Subject: [PATCH] Fix gnmi_cli hang
+
```
Comment (ganglyu) [inline patches/0001-Fix-gnmi_cli-hang.patch:5]: Please add comment to explain your patch

File: Makefile [hunk 2]
```diff
@@ -53,6 +53,7 @@ $(GO_DEPS): go.mod $(PATCHES) swsscommon_wrap
 	patch -d vendor -p0 < patches/gnmi_path.patch
 	patch -d vendor -p0 < patches/gnmi_xpath.patch
 	git apply patches/0001-Updated-to-filter-and-write-to-file.patch
+	git apply patches/0001-Fix-gnmi_cli-hang.patch
```
Comment (renukamanavalan) [inline Makefile:56]: This should be "0002-...."

**Review by renukamanavalan — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (zbud-msft) [reply Makefile]: Combined into one patch

File: patches/0001-Fix-gnmi_cli-hang.patch (hunk 1 above)
Comment (zbud-msft) [reply patches/0001-Fix-gnmi_cli-hang.patch]: Fixed last patch by removing writing to channel inside cli_config. Combined fixes into existing patch

**Review by ganglyu — APPROVED** [review]

**Pull request merged**